addons/stock/models/stock_move.py

```diff
@@ -490,7 +490,7 @@ def _set_lot_ids(self):
                 mls = move.move_line_ids
             mls = mls.filtered(lambda ml: ml.lot_id)
             for ml in mls:
-                if ml.lot_id not in move.lot_ids:
```

Comment:
why not
```python
if ml.lot_id and ml.lot_id not in move.lot_ids
```

Reply:
Because `move.lot_ids` will be empty

After the user modified the produced quantity (step 6), when writing the `onchange` results, the server first computes the stock move's lot ids: https://github.com/odoo/odoo/blob/f8861e6a63fcd85295affdea4b525af753234b14/addons/stock/models/stock_move.py#L470-L482 and here is the problem: `domain_nosuggest` and `domain_suggest` do not consider the case where a ML has some quantities that are only reserved. Therefore, after that, when the server updates the stock move, it will delete the ML since `move.lot_ids` is empty : https://github.com/odoo/odoo/blob/f8861e6a63fcd85295affdea4b525af753234b14/addons/stock/models/stock_move.py#L492-L494

This is the reason why I was first wondering if the fix could be a modification of `domain_nosuggest` and/or  `domain_suggest` so `move.lot_ids` will not be empty.

Reply:
Ok du coup. if ml.quantity_done and ml.lot_id not in move.lot_ids: